src/site/stages/build/process-cms-exports/transform-fields.js

```diff
@@ -0,0 +1,148 @@
+ */
+function transformFields(entity, outputSchemaFromParent) {
+  // console.log(JSON.stringify(entity, null, 2));
+  // throw new Error('stopping');
+
+  const inputSchema = getInputSchema(entity);
+  if (!inputSchema) {
+    throw new Error(
+      `Could not find input schema for ${getContentModelType(entity)}`,
+    );
+  }
+
+  const outputSchema = outputSchemaFromParent || getOutputSchema(entity);
+  if (!outputSchema) {
+    // eslint-disable-next-line no-console
+    console.log(`No output schema found for ${getContentModelType(entity)}`);
+    return entity;
+  }
+
+  // Iterate over each field:
+  return Object.entries(outputSchema.properties).reduce(
+    /* eslint-disable no-param-reassign */
+    /* eslint-disable no-console */
+    (result, [outputKey, outputFieldSchema]) => {
+      // Find the snake_case key for use in the input schema and entity data
+      const inputKey = _.snakeCase(outputKey);
+
+      // If the input schema is an entity reference, find the entity reference data and recurse
+      if (isEntityReferenceArray(entity[inputKey])) {
+        console.log(
+          chalk.gray(`Expanding ${outputKey} (${getContentModelType(entity)})`),
+        );
+        result[outputKey] = entity[inputKey].map(ref => {
+          const childEntity = readEntity(
+            path.resolve(
+              __dirname,
+              '../../../../../.cache/localhost/cms-export-content',
+            ),
+            ref.target_type,
+            ref.target_uuid,
+          );
+          return transformFields(childEntity, outputFieldSchema);
+        });
+
+        if (
+          result[outputKey].length > 1 &&
+          outputFieldSchema.type !== 'array'
+        ) {
+          // eslint-disable-next-line no-console
+          console.error(
+            chalk.red(
+              `Multiple entity references found at ${inputKey} on ${getContentModelType(
+                entity,
+              )}, but the output schema for ${outputKey} expects a single object.`,
+            ),
+          );
+        } else if (outputFieldSchema.type === 'object') {
+          result[outputKey] = result[outputKey][0];
+        }
+        return result;
+      }
+
+      // TODO: If the content model has a transformer hook for that field, use it
+
+      // Find the input schema transformers
+      const inputTransformerMap = fieldTransformers.get(
+        serialize(inputSchema.properties[inputKey]),
+      );
+      if (!inputTransformerMap) {
+        console.warn(
+          chalk.yellow(
+            `Could not find transformer map for ${inputKey} (${getContentModelType(
+              entity,
+            )})`,
+          ),
+        );
+        result[outputKey] = entity[inputKey];
+        return result;
+      }
+
+      // Find the specific field transformer
+      const transformField = inputTransformerMap.get(
+        serialize(outputSchema.properties[outputKey]),
+      );
```

Comment:
I don't quite understand the situation where that inner `Map` has multiple mappings that we care about.